### PR TITLE
src: Integrate image-builder-frontend as a Cockpit package with webpack support

### PR DESCRIPTION
**Pull request by mgold1234**
Screenshot 2024-09-04 at 18 05 58

- Added image-builder-frontend repository as a Cockpit package.     
- Configured webpack with necessary loaders and plugins for handling JS, TS, CSS, and SCSS files.
- Ensured compatibility with Cockpit by externalizing cockpit dependencies and updating the webpack configuration.
- Provided initial setup with image-builder-frontend.

### CODE REVIEW

File: public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "requires": {
+    "cockpit": "134"
+  },
+  "dashboard": {
+    "index": {
+      "label": "Images",
```
Comment (croissanne) [inline public/manifest.json:10]: I'd make this Image Builder, it's set to replace cockpit-composer after all.

File: src/AppCockpit.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                  Image builder for RPM-DNF documentation
+                </Button>
+              </EmptyStateActions>
+            </EmptyStateFooter>
+          </>
+        </EmptyState>
+      </Bullseye>
+      <NotificationsPortal />
+      <BrowserRouter>
+        <Routes>
```
Comment (croissanne) [inline src/AppCockpit.tsx:90]: I think we should reuse the routes from `Router.tsx`, and just have the BrowserRouter component in here.

File: package.json [hunk 3]
```diff
@@ -92,6 +92,7 @@
     "test:watch": "TZ=UTC vitest",
     "test:coverage": "TZ=UTC vitest run --coverage",
     "build": "fec build",
+    "build:dev": "webpack --config webpack.config.js",
```
Comment (croissanne) [inline package.json:95]: Can we rename cockpit related commands so it's obvious? Like "cockpit:build" etc.

**Review by croissanne — CHANGES_REQUESTED** [review]
Neat! Thank you.

I wonder if the directory structure can be nicer if we put the cockpit related files into it's own directory somehow, like src/cockpit/AppCockpit.tsx, and `cockpit/public / cockpit/webpack.config.js` ?

Also would it be good to reuse a single existing component as a poc in this first version? Like anything, the landing page or something that doesn't require any api interaction.
I

File: public/manifest.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "requires": {
+    "cockpit": "134"
+  },
+  "version": 0,
+  "name": "composer",
```
Comment (KKoukiou) [inline public/manifest.json:6]: Decide what URL you want to use. If it's images in the path you want to change that.

File: public/manifest.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  "priority": 50,
```
Comment (KKoukiou) [inline public/manifest.json:7]: You can switch to priority: 0 if you use a new name. Aka not composer.

**Review by KKoukiou — COMMENTED** [review]
You will need to create a Fedora package, aka you 're missing a spec file at least here.

In that you want to obsolete the old package. https://docs.fedoraproject.org/en-US/packaging-guidelines/#renaming-or-replacing-existing-packages

File: src/AppCockpit.tsx (hunk 2 above)
Comment (mgold1234) [reply src/AppCockpit.tsx:90]: I am sure its douable in that stage, becuase Router.txs require any api interaction.
@lucasgarfield WDYT?

**Pull request closed without merging**